src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

```diff
@@ -0,0 +1,2271 @@
+		JPanel panel = new JPanel(new GridLayout(1, 2));
+		panel.add(new JLabel(MessageBundle.getMessage("angal.medicalstock.multiplecharging.expiringdate")));
+		panel.add(expireDateChooser);
+
+		int ok = JOptionPane.showConfirmDialog(this, panel,
+			MessageBundle.getMessage("angal.medicalstock.multiplecharging.expiringdate"),
+			JOptionPane.OK_CANCEL_OPTION);
+
+		if (ok == JOptionPane.OK_OPTION) {
+			date = expireDateChooser.getLocalDateTime();
+		}
+		return date;
+	}
+
+	protected double askTotalCost() {
+		String input = JOptionPane.showInputDialog(this,
+			MessageBundle.getMessage("angal.medicalstock.multiplecharging.totalcost"),
+			0.);
+		double total = 0.;
+		if (input != null) {
+			try {
+				total = Double.parseDouble(input);
+				if (total < 0) {
+					throw new NumberFormatException();
+				}
+			} catch (NumberFormatException nfe) {
+				MessageDialog.error(this, "angal.medicalstock.multiplecharging.pleaseinsertavalidvalue");
+			}
+		}
+		return total;
+	}
+
+	public MedicalInventory getInventory() {
+		return inventory;
+	}
+
+	public void setInventory(MedicalInventory inventory) {
+		this.inventory = inventory;
+	}
+
+	private JLabel getSpecificRadio() {
+		if (specificRadio == null) {
+			specificRadio = new JLabel(MessageBundle.getMessage("angal.inventory.specificproduct.txt"));
+		}
+		return specificRadio;
+	}
+
+	private JLabel getDateInventoryLabel() {
+		if (dateInventoryLabel == null) {
+			dateInventoryLabel = new JLabel(MessageBundle.getMessage("angal.common.date.txt"));
+		}
+		return dateInventoryLabel;
+	}
+
+	private JTextField getCodeTextField() {
+		if (codeTextField == null) {
+			codeTextField = new JTextField();
+			codeTextField.setColumns(10);
+			TextPrompt suggestion = new TextPrompt(MessageBundle.getMessage("angal.common.code.txt"), codeTextField, Show.FOCUS_LOST);
+			suggestion.setFont(new Font("Tahoma", Font.PLAIN, 12));
+			suggestion.setForeground(Color.GRAY);
+			suggestion.setHorizontalAlignment(JLabel.CENTER);
+			suggestion.changeAlpha(0.5f);
+			suggestion.changeStyle(Font.BOLD + Font.ITALIC);
+			codeTextField.addKeyListener(new KeyAdapter() {
+
+				@Override
+				public void keyPressed(KeyEvent e) {
+					if (e.getKeyCode() == KeyEvent.VK_ENTER) {
+						code = codeTextField.getText().trim();
+						code = code.toLowerCase();
+						try {
+							addInventoryRow(code);
+						} catch (OHServiceException e1) {
+							OHServiceExceptionUtil.showMessages(e1);
+						}
+						if (inventory != null && !inventory.getStatus().equals(InventoryStatus.draft.toString())) {
+							inventory.setStatus(InventoryStatus.draft.toString());
+						}
+						codeTextField.setText("");
+					}
+				}
+			});
+		}
+		return codeTextField;
+	}
+
+	private List<MedicalInventoryRow> loadNewInventoryTable(String code, MedicalInventory inventory, boolean add) throws OHServiceException {
+		List<MedicalInventoryRow> inventoryRowsList = new ArrayList<>();
```

Comment:
```suggestion
		List<MedicalInventoryRow> inventoryRowsList;
```
Useless assignment as all code paths set a value to `inverntoryRowsList`